File: Leetcode_starter/leetcode009.cpp
/*
 * leetcode 1009
 * 不将整数转为字符串来判断一个整数是不是回文数
 * 要点是找技巧，经典方法是通过除法与求余将数字从尾部依次倒转，只截一半
 */

class Solution {
public:
    bool isPalindrome(int x) {
        int ans = 0;
        int cur;
        while(x > ans)
        {
            cur = x % 10;
            ans = ans * 10 + cur;
            x = x / 10;
        }
        return ans == x || ans == x * 10
    }
};

File: Leetcode_starter/leetcode009.py
## 1009
## 回文数
## 要点是找到方法，都截取到中间，从后往前的数字要和从前往后的一样

class Solution:
    def isPalindrome(self, x: int) -> bool:
        reverse = 0
        while x > reverse:
            reverse = reverse * 10 + x % 10
            x = x // 10

        return(x == reverse or reverse // 10 == x)


if __name__ == '__main__':
    s = Solution()
    print(s.isPalindrome(12321))